### PR TITLE
Simplify withdrawals logic

### PR DESCRIPTION
**Pull request by Nashtare**
cf https://github.com/0xPolygonZero/zk_evm/issues/167

### CODE REVIEW

**Comment by Nashtare** [comment]
Hmm I have no idea why the CI is failing to kick off though 🤔

**Comment by muursh** [comment]
Lol, I've just opened this and seen that you did CI stuff here. #17 is probably superfluous

**Comment by Nashtare** [comment]
It's fine we can merge it and I'll pull :) I was also bothered by the `cI` typo

**Review by muursh — APPROVED** [review]
lgtm :)

File: leader/src/padding_and_withdrawals.rs [hunk 1]
```diff
@@ -109,50 +99,17 @@ pub(crate) fn add_withdrawals_to_txns(
...
+        .last_mut()
+        .expect("We cannot have an empty list of payloads.");
 
-            // If we have dummy proofs (note: `txn_ir[1]` is always a dummy txn in this
-            // case), then this dummy will get the withdrawals.
-            txn_ir[1].withdrawals = withdrawals;
-            txn_ir[1].trie_roots_after.state_root = final_trie_state.state.hash();
-        }
-    }
+    last_inputs.withdrawals = withdrawals;
```
Comment (muursh) [inline leader/src/padding_and_withdrawals.rs:111]: I'll be honest this made me double take at first

**Pull request merged**